internal/pkg/service/stream/definition/repository/sink/sink_repo.go

```diff
@@ -0,0 +1,103 @@
+package sink
+
+import (
+	"context"
+	"time"
+
+	"github.com/keboola/go-utils/pkg/deepcopy"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/serde"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/repository/sink/schema"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/repository/source"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/plugin"
+)
+
+type Repository struct {
+	client  etcd.KV
+	schema  schema.Sink
+	plugins *plugin.Plugins
+	sources *source.Repository
+}
+
+type dependencies interface {
+	EtcdClient() *etcd.Client
+	EtcdSerde() *serde.Serde
```

Comment:
Yes, it's definitely the perfect time.

- `*etcd.Client` if standard client from the `etcd` pkg.
- `serde` is separated, because different prefixes can have different encoding if needed.
  - If there is a problem with size/performance, it is possible to use binary formats, e.g. ProtocolBuffers.